File: python-wrapper/src/python/model/embed/TelecomType.py
from typing import Union, Dict
from enum import Enum

class TelecomType(Enum):
	Mobile = "mobile"
	Phone = "phone"
	Email = "email"
	Fax = "fax"
	Skype = "skype"
	Im = "im"
	Medibridge = "medibridge"
	Ehealthbox = "ehealthbox"
	Apicrypt = "apicrypt"
	Web = "web"
	Print = "print"
	Disk = "disk"
	Other = "other"
	Pager = "pager"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'TelecomType':
		if data == "mobile":
			return TelecomType.Mobile
		elif data == "phone":
			return TelecomType.Phone
		elif data == "email":
			return TelecomType.Email
		elif data == "fax":
			return TelecomType.Fax
		elif data == "skype":
			return TelecomType.Skype
		elif data == "im":
			return TelecomType.Im
		elif data == "medibridge":
			return TelecomType.Medibridge
		elif data == "ehealthbox":
			return TelecomType.Ehealthbox
		elif data == "apicrypt":
			return TelecomType.Apicrypt
		elif data == "web":
			return TelecomType.Web
		elif data == "print":
			return TelecomType.Print
		elif data == "disk":
			return TelecomType.Disk
		elif data == "other":
			return TelecomType.Other
		elif data == "pager":
			return TelecomType.Pager
		else:
			raise Exception(f"{data} is not a valid value for TelecomType enum.")

__all__ = ['TelecomType']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/TelecomType.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class TelecomType(
	internal val dtoSerialName: String,
) {
	@SerialName("mobile")
	Mobile("mobile"),

	@SerialName("phone")
	Phone("phone"),

	@SerialName("email")
	Email("email"),

	@SerialName("fax")
	Fax("fax"),

	@SerialName("skype")
	Skype("skype"),

	@SerialName("im")
	Im("im"),

	@SerialName("medibridge")
	Medibridge("medibridge"),

	@SerialName("ehealthbox")
	Ehealthbox("ehealthbox"),

	@SerialName("apicrypt")
	Apicrypt("apicrypt"),

	@SerialName("web")
	Web("web"),

	@SerialName("print")
	Print("print"),

	@SerialName("disk")
	Disk("disk"),

	@SerialName("other")
	Other("other"),

	@SerialName("pager")
	Pager("pager"),
}
